### PR TITLE
Improves some turret code and allows access customization 

### PR DESCRIPTION
**Pull request by XDTM**
:cl: XDTM
add: Portable Turrets (the ones that the AI uses) now require turret electronics to be built.
add: Turret electronics can be built at any autolathe or at the engineering or security protolathes.
add: Turret electronics can be used inhand to customize the turret's access requirements and initial settings.
add: Turret control panels now start without any access requirements; swiping an ID will copy its access requirements.
add: Turret frames now have hints during construction.
fix: The Neutralize Identified Criminals setting now actually does something. Check Security Records has been removed since it did the same thing.
fix: Checking weapon authentication will no longer prevent scanning for mindshield or arrest status if the target has turret access.
/:cl:

this code makes me cry a bit less now

### CODE REVIEW

File: code/game/machinery/porta_turret/portable_turret_construct.dm [hunk 1]
```diff
@@ -17,7 +18,30 @@
 	density = TRUE
 	var/build_step = PTURRET_UNSECURED //the current step in the building process
 	var/finish_name = "turret"	//the name applied to the product turret
+	var/obj/item/electronics/turret/electronics
 	var/obj/item/gun/installed_gun = null
+	
+/obj/machinery/porta_turret_construct/examine(mob/user)
```
Comment (MrDoomBringer) [inline code/game/machinery/porta_turret/portable_turret_construct.dm]: Could you add hints for the deconstruction of this as well? like in reinf_walls.dm

**Comment by creeperparty568** [comment]
> add: Turret electronics can be used inhand to customize the turret's access requirements and initial settings.
add: Turret control panels now start without any access requirements; swiping an ID will copy its access requirements.

I don't think that it's the best idea to allow _anyone_ to make **possibly lethal turrets**. Engineering and Security protolathes make more sense as is having turret access be sec/engineering but we don't want greyshirts making turrets.

File: code/game/machinery/porta_turret/portable_turret_construct.dm [hunk 2]
```diff
@@ -192,4 +231,7 @@
 	if(installed_gun)
 		qdel(installed_gun)
```
Comment (ShizCalev) [inline code/game/machinery/porta_turret/portable_turret_construct.dm]: QDEL_NULL

File: code/game/machinery/porta_turret/portable_turret_construct.dm [hunk 3]
```diff
@@ -192,4 +231,7 @@
...
 		installed_gun = null
+	if(electronics)
+		qdel(electronics)
```
Comment (ShizCalev) [inline code/game/machinery/porta_turret/portable_turret_construct.dm]: QDEL_NULL

**Review by ShizCalev — CHANGES_REQUESTED** [review]

File: code/game/machinery/porta_turret/portable_turret.dm [hunk 4]
```diff
@@ -53,12 +55,11 @@
...
-	var/stun_all = 0		//if this is active, the turret shoots everything that isn't security or head of staff
-	var/check_anomalies = 1	//checks if it can shoot at unidentified lifeforms (ie xenos)
-	var/shoot_unloyal = 0	//checks if it can shoot people that aren't loyalty implantd
+	var/check_arrest = TRUE		//checks if it can shoot people under arrest
+	var/check_weapons = FALSE	//checks if it can shoot people that have a weapon they aren't authorized to have
+	var/stun_all = FALSE		//if this is active, the turret shoots everything that isn't authorized
+	var/check_anomalies = TRUE	//checks if it can shoot at unidentified lifeforms (ie xenos)
+	var/check_mindshield = FALSE	//checks if it can shoot people that aren't mindshield implantd
 
 	var/attacked = 0		//if set to 1, the turret gets pissed off and shoots at people nearby (unless they have sec access!)
```
Comment (ExcessiveUseOfCobblestone) [inline code/game/machinery/porta_turret/portable_turret.dm]: looks like a boo as well

**Comment by ExcessiveUseOfCobblestone** [comment]
one area of improvement i'd like to explore is to convert check anom into a bit so each item within it can be toggled on off, allowing for far greater control over the turret (perhaps you want to turn off alien check since the turret resides in xenobio or something).

**Comment by Cyprex** [comment]
Thank you for improving turrets! Does this PR allow linking turrets to a control panel?

**Pull request closed without merging**